Classes/commands.py

```diff
@@ -1029,3 +1032,78 @@ async def team_json(self, ctx):
 
         # Send the JSON file
         await send_long(ctx.channel, json.dumps(json_out), code_block=True)
+
+    @checks.is_admin_bot_channel()
+    @checks.is_white_shirt()
+    @commands.command()
+    # Reimplementation of old ?count_officers command
+    async def count_officers(self, ctx):
+        # Call our function to get a list of roles
+        role_ids = role_id_index(self)
+
+        # Build index of Officers, keeping only the highest role in the ladder
+        all_officers = []
+        guild = self.bot.officer_manager.guild
+        for member in guild.members:
+            for role in member.roles:
+                if role.id in role_ids:
+                    if member in all_officers:
+                        del all_officers[-1]
+                    all_officers.append(member)
+
+        # Get a usable number of oficers, and create a dictionary for the count by role
+        number_of_officers = len(all_officers)
+        number_of_officers_with_each_role = {}
+
+        # For every role in the role list, reverse sorted to preserve higher role:
+        for entry in role_ids[::-1]:
+            role = guild.get_role(entry)
+            if (
+                role is None
+            ):  # If the role ID is invalid, let the user know what the role name should be, and that the ID in settings is invalid
+                await ctx.channel.send(
+                    ctx.message.author.mention
+                    + " The role ID for "
+                    + role_id_index(self, entry)
+                    + " has been corrupted in the bot configuration, therefore I cannot provide an accurate count. Please alert the Programming Team. Displayed below are the results of counting all other roles."
+                )
```

Comment:
This can be converted to an f-string to both increase readability and avoid crashes when you add numbers and strings.

Reply:
Fixed in commit